src/app-logic/url-handling.js

```diff
@@ -650,12 +710,13 @@ function convertHiddenLocalTracksByPidFromString(
   const hiddenLocalTracksByPid = new Map();
 
   for (const stringPart of rawText.split('~')) {
-    const [pidString, ...indexStrings] = stringPart.split('-');
-    if (indexStrings.length === 0) {
+    if (stringPart.indexOf('-') === -1) {
       continue;
     }
+    const pidString = stringPart.slice(0, stringPart.indexOf('-'));
+    const hiddenTracksString = stringPart.slice(pidString.length + 1);
```

Comment:
optional: I feel like you could store `stringPart.indexOf('-')` in a variable once and use that for the `if` condition and the 2 slices.

Reply:
You can also use `split` with the second optional argument `limit` set to `2`, which may be clearer?
(still optional)

Reply:
(same comment for `convertLocalTrackOrderByPidFromString`)

Reply:
I tried using `split` with `limit` set to `2`, but it wasn't doing what I needed: `'2-1-5-4'.split('-', 2)` is `['2', '1']`, but I'd need it to be `['2', '1-5-4']`.

Reply:
Thanks for trying, clearly I misread the documentation. We both learnt something today :-)